opm/test_util/EclFilesComparator.cpp

```diff
@@ -20,173 +20,105 @@
-            return false;
-        default:
-            std::cout << "\nKeyword " << keyword << "has undefined type." << std::endl;
-            return false;
-    }
-}
-
-
-
-void ECLFilesComparator::printResultsForKeyword(const std::string& keyword) const {
-    const double absDeviationAverage = average(absDeviation);
-    const double relDeviationAverage = average(relDeviation);
-    std::cout << "Average absolute deviation = " << absDeviationAverage  << std::endl;
-    std::cout << "Median absolute deviation  = " << median(absDeviation) << std::endl;
-    std::cout << "Average relative deviation = " << relDeviationAverage  << std::endl;
-    std::cout << "Median relative deviation  = " << median(relDeviation) << "\n\n";
 }
 
 
-
-void ECLFilesComparator::deviationsForOccurence(const std::string& keyword, int occurence) {
-    ecl_kw_type* ecl_kw1         = ecl_file_iget_named_kw(ecl_file1, keyword.c_str(), occurence);
-    ecl_kw_type* ecl_kw2         = ecl_file_iget_named_kw(ecl_file2, keyword.c_str(), occurence);
+unsigned int ECLFilesComparator::getEclKeywordData(ecl_kw_type*& ecl_kw1, ecl_kw_type*& ecl_kw2, const std::string& keyword, int occurrence1, int occurrence2) const {
+    ecl_kw1 = ecl_file_iget_named_kw(ecl_file1, keyword.c_str(), occurrence1);
+    ecl_kw2 = ecl_file_iget_named_kw(ecl_file2, keyword.c_str(), occurrence2);
     const unsigned int numCells1 = ecl_kw_get_size(ecl_kw1);
     const unsigned int numCells2 = ecl_kw_get_size(ecl_kw2);
     if (numCells1 != numCells2) {
-        std::cout << "For keyword " << keyword << " and occurence " << occurence << ":\n";
-        OPM_THROW(std::runtime_error, "Number of active cells are different.");
-    }
-    std::vector<double> values1(numCells1), values2(numCells2);
-    ecl_kw_get_data_as_double(ecl_kw1, values1.data());
-    ecl_kw_get_data_as_double(ecl_kw2, values2.data());
-
-    auto it = std::find(keywordWhitelist.begin(), keywordWhitelist.end(), keyword);
-    for (unsigned int cell = 0; cell < values1.size(); cell++) {
-        deviationsForCell(values1[cell], values2[cell], keyword, occurence, cell, it == keywordWhitelist.end());
+        OPM_THROW(std::runtime_error, "For keyword " << keyword << ":"
+                << "\nOccurrence in first file " << occurrence1
+                << "\nOccurrence in second file " << occurrence2
+                << "\nCells in first file: " << numCells1
+                << "\nCells in second file: " << numCells2
+                << "\nThe number of cells differ.");
     }
+    return numCells1;
 }
 
 
 
-void ECLFilesComparator::deviationsForCell(double val1, double val2, const std::string& keyword, int occurence, int cell, bool allowNegativeValues) {
+template <typename T>
+void ECLFilesComparator::printValuesForCell(const std::string& keyword, int occurrence1, int occurrence2, size_t cell, const T& value1, const T& value2) const {
     int i, j, k;
     ecl_grid_get_ijk1(ecl_grid1, cell, &i, &j, &k);
```

Comment:
>> We actually have no way of knowing that.
>
> As I said - probably.

True.  On the other hand, since we do not know what function to call, that is a sign that there is a possibly a piece of information missing at the call site.